Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -0,0 +1,392 @@
+        glyphs->_baselineOriginY.push_back(baselineOriginY);
+
+        _DWriteGlyphRun glyphRunInfo;
+        glyphRunInfo._glyphCount = glyphRun->glyphCount;
+        for (int index = 0; index < glyphRun->glyphCount; index++) {
+            glyphRunInfo._glyphIndices.push_back(glyphRun->glyphIndices[index]);
+            glyphRunInfo._glyphAdvances.push_back(CGSizeMake(glyphRun->glyphAdvances[index], 0));
+        }
+        glyphs->_glyphRuns.push_back(glyphRunInfo);
+
+        _DWriteGlyphRunDescription glyphRunDescriptionInfo;
+        glyphRunDescriptionInfo._stringLength = glyphRunDescription->stringLength;
+        glyphRunDescriptionInfo._textPosition = glyphRunDescription->textPosition;
+        glyphs->_glyphRunDescriptions.push_back(glyphRunDescriptionInfo);
+
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE DrawUnderline(_In_ void* clientDrawingContext,
+                                            _In_ float baselineOriginX,
+                                            _In_ float baselineOriginY,
+                                            _In_ DWRITE_UNDERLINE const* underline,
+                                            _In_ IUnknown* clientDrawingEffects) throw() {
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE DrawStrikethrough(_In_ void* clientDrawingContext,
+                                                _In_ float baselineOriginX,
+                                                _In_ float baselineOriginY,
+                                                _In_ DWRITE_STRIKETHROUGH const* strikethrough,
+                                                _In_ IUnknown* clientDrawingEffects) throw() {
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE DrawInlineObject(_In_ void* clientDrawingContext,
+                                               _In_ float originX,
+                                               _In_ float originY,
+                                               _In_ IDWriteInlineObject* inlineObject,
+                                               _In_ BOOL isSideways,
+                                               _In_ BOOL isRightToLeft,
+                                               _In_ IUnknown* clientDrawingEffects) throw() {
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE IsPixelSnappingDisabled(_In_opt_ void* clientDrawingContext, _Out_ BOOL* isDisabled) throw() {
+        *isDisabled = false;
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE GetCurrentTransform(_In_opt_ void* clientDrawingContext, _Out_ DWRITE_MATRIX* transform) throw() {
+        return S_OK;
+    };
+
+    HRESULT STDMETHODCALLTYPE GetPixelsPerDip(_In_opt_ void* clientDrawingContext, _Out_ float* pixelsPerDip) throw() {
+        *pixelsPerDip = 1.0;
+        return S_OK;
+    };
+
+private:
+    ComPtr<IDWriteFactory> _dwriteFactory;
+};
+
+CustomTargetTextRenderer::CustomTargetTextRenderer() {
+    _dwriteFactory = GetDWriteFactoryInstance();
+}
```

Comment:
Yes this code is not used any more. Will remove

---
In reply to: [77756443](https://github.com/Microsoft/WinObjC/pull/950#discussion_r77756443) [](ancestors = 77756443)